pkg/envoy/configurator.go

```diff
@@ -25,60 +33,123 @@ type KubernetesConfigurator struct {
 }
 
 //NewKubernetesConfigurator returns a Kubernetes configurator given a lister and ingress class
-func NewKubernetesConfigurator(lister k8s.IngressLister, nodeID, cert, key, ca string, ingressClasses []string) *KubernetesConfigurator {
-	return &KubernetesConfigurator{lister: lister, ingressClasses: ingressClasses, nodeID: nodeID, cert: cert, key: key, trustCA: ca}
+func NewKubernetesConfigurator(nodeID string, certificates []Certificate, ca string, ingressClasses []string) *KubernetesConfigurator {
+	return &KubernetesConfigurator{ingressClasses: ingressClasses, nodeID: nodeID, certificates: certificates, trustCA: ca}
 }
 
 //Generate creates a new snapshot
-func (c *KubernetesConfigurator) Generate() (cache.Snapshot, error) {
-	ingresses, err := c.lister.List()
+func (c *KubernetesConfigurator) Generate(ingresses []v1beta1.Ingress) cache.Snapshot {
```

Comment:
This doesn't change the fact that it was handling ingress objects. That is what is returned by `c.lister.List()`.

I made this change so we could test the purpose of this function. Ingress -> Envoy Config.

Previously it would have been a pain to test this Configurator as it would mean mocking out the Lister so that you could control what it returned inside of this function. By shifting the Lister to the Snapshot it makes this so much easier to write tests for.